command/agent.go

```diff
@@ -431,15 +431,15 @@ func (cmd *AgentCommand) readConfig() *agent.Config {
 	}
 
 	// Need both tag key and value for EC2 discovery
-	if cfg.RetryJoinEC2.TagKey != "" || cfg.RetryJoinEC2.TagValue != "" {
-		if cfg.RetryJoinEC2.TagKey == "" || cfg.RetryJoinEC2.TagValue == "" {
+	if cfg.DeprecatedRetryJoinEC2.TagKey != "" || cfg.DeprecatedRetryJoinEC2.TagValue != "" {
+		if cfg.DeprecatedRetryJoinEC2.TagKey == "" || cfg.DeprecatedRetryJoinEC2.TagValue == "" {
 			cmd.UI.Error("tag key and value are both required for EC2 retry-join")
 			return nil
 		}
 	}
 
 	// EC2 and GCE discovery are mutually exclusive
-	if cfg.RetryJoinEC2.TagKey != "" && cfg.RetryJoinEC2.TagValue != "" && cfg.RetryJoinGCE.TagValue != "" {
+	if cfg.DeprecatedRetryJoinEC2.TagKey != "" && cfg.DeprecatedRetryJoinEC2.TagValue != "" && cfg.DeprecatedRetryJoinGCE.TagValue != "" {
 		cmd.UI.Error("EC2 and GCE discovery are mutually exclusive. Please provide one or the other.")
```

Comment:
We should be able to delete this and support multiple. Not that people would do this, but it doesn't need this nasty special case check.